### PR TITLE
Python: extended notebook 06 to showcase Azure AI Search for semantic memory

### PR DESCRIPTION
**Pull request by rom212**
### Motivation and Context

Sample notebook 06-memory-and-embeddings demonstrated how semantic kernel lets users add memory to their kernels and also work around ever-growing contexts that eventually exceed the maximum token limit. It did so by using the in RAM VolatileMemoryStore and teased the possibility of offloading this to an external vector store. With this pull request, we want to provide guidance and repeatable steps so that users can actually perform semantic search by leveraging an external vector store, Azure AI Search (formerly known as Azure Cognitive Search

### Description

The changes consist in new cells added at the tail end of the existing notebook. We register a new memory store, external this time and show case how the rest of the code can be reused as-is. We add the capability of providing access credentials to Azure AI Search in environment variables as well.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by rom212** [comment]
> @microsoft-github-policy-service agree [company="microsoft"]

**Review by moonbox3 — COMMENTED** [review]
@rom212 Since you're already updating this notebook, can you please remove the unnecessary "a" in the following sentence in the intro text above the first cell?

Sentence to modify, please: "... would grow so large that you would run into ~~a~~ the model's token limit."

**Comment by rom212** [comment]
> @rom212 Since you're already updating this notebook, can you please remove the unnecessary "a" in the following sentence in the intro text above the first cell?
> 
> Sentence to modify, please: "... would grow so large that you would run into ~a~ the model's token limit."

Done. Sorry for missing it.

File: python/notebooks/06-memory-and-embeddings.ipynb [hunk 1]
```diff
@@ -390,9 +515,126 @@
...
+      "cell_type": "code",
+      "execution_count": 13,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "from semantic_kernel.connectors.memory.azure_cognitive_search import AzureCognitiveSearchMemoryStore\n",
         "\n",
-        "Stay tuned for that!"
+        "azure_ai_search_api_key, azure_ai_search_url = sk.azure_aisearch_settings_from_dot_env()\n",
+        "kernel.register_memory_store(memory_store=AzureCognitiveSearchMemoryStore(1536,search_endpoint=azure_ai_search_url,admin_key=azure_ai_search_api_key))\n"
```
Comment (moonbox3) [inline python/notebooks/06-memory-and-embeddings.ipynb:530]: For the `vector_size` = 1536, could we use the named argument, please? It helps clarify for the reader/user if they're not familiar with what "1536" is.

Comment (moonbox3) [reply python/notebooks/06-memory-and-embeddings.ipynb:530]: My other small request with this line would be to try and create some line wrapping, please? Something like this:

kernel.register_memory_store(
    memory_store=AzureCognitiveSearchMemoryStore(
        vector_size=1536,
        search_endpoint=azure_ai_search_url,
        admin_key=azure_ai_search_api_key
    )
)

File: python/notebooks/06-memory-and-embeddings.ipynb [hunk 2]
```diff
@@ -390,9 +515,126 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "We can see that our function created an \"About Me\" index and that our five pieces of infomation have been indexed (note that it can take a few minutes for the UI to reflect the document count and storage size)."
+      ]
+    },
+    {
+      "attachments": {
+        "image.png": {
```
Comment (moonbox3) [inline python/notebooks/06-memory-and-embeddings.ipynb:566]: I am having a difficult time reading the text in the images. Perhaps we could crop the image and show the most relevant part? This goes for the second image as well.

File: python/semantic_kernel/utils/settings.py [hunk 3]
```diff
@@ -230,3 +230,18 @@ def redis_settings_from_dot_env() -> str:
     ), "Redis connection string not found in .env file"
 
     return connection_string
+
+def azure_aisearch_settings_from_dot_env() -> Tuple[str, str]:
+    """
+    Reads the Azure AI Search environment variables for the .env file.
+
+    Returns:
```
Comment (moonbox3) [inline python/semantic_kernel/utils/settings.py:238]: I'd love to please include exactly what is returned (e.g., the Azure AI Search API Key and the Azure AI Search URL).

**Comment by moonbox3** [comment]
Thank you for extending this notebook, @rom212. I suggest, after addressing the feedback, to commit the notebook without running all of the cells. All other example notebooks have been committed without running all.

**Comment by moonbox3** [comment]
> > @rom212 Since you're already updating this notebook, can you please remove the unnecessary "a" in the following sentence in the intro text above the first cell?
> > Sentence to modify, please: "... would grow so large that you would run into ~a~ the model's token limit."
> 
> Done. Sorry for missing it.

No worries at all! Thank you for doing it. Figured it'd be great to do the cleanup since you're already working with the file.

File: python/notebooks/06-memory-and-embeddings.ipynb (hunk 1 above)
Comment (rom212) [reply python/notebooks/06-memory-and-embeddings.ipynb]: Hi Evan, I hope this is addressed in  commit 957f3d7d52d05075b1c524f32c5e037713dc73ae

File: python/notebooks/06-memory-and-embeddings.ipynb (hunk 2 above)
Comment (rom212) [reply python/notebooks/06-memory-and-embeddings.ipynb:459]: Hi Evan, I hope this is addressed to your satisfaction in commit 957f3d7d52d05075b1c524f32c5e037713dc73ae

File: python/semantic_kernel/utils/settings.py [hunk 4]
```diff
@@ -230,3 +230,18 @@ def redis_settings_from_dot_env() -> str:
...
+        dict: The Azure AI Search environment variables
+    """
+    config = dotenv_values(".env")
+    api_key = config.get("AZURE_AISEARCH_API_KEY", None)
```
[the remark this answers is not in the document]
Comment (rom212) [reply python/semantic_kernel/utils/settings.py:242]: Makes sense. added in commit 957f3d7d52d05075b1c524f32c5e037713dc73ae

File: python/semantic_kernel/utils/settings.py (hunk 3 above)
Comment (rom212) [reply python/semantic_kernel/utils/settings.py:238]: Makes sense as well, updated in commit 957f3d7d52d05075b1c524f32c5e037713dc73ae

**Comment by rom212** [comment]
> Thank you for extending this notebook, @rom212. I suggest, after addressing the feedback, to commit the notebook without running all of the cells. All other example notebooks have been committed without running all.

Ok Did a "restart" and "Clear All Outputs". Will keep this in mind.

**Comment by moonbox3** [comment]
Hi @rom212, can you please fix the following typos so that the proper checks can run successfully?

 "We can see that our function created an \"About Me\" index and that our five pieces of **<ins>infomation</ins>** have been indexed (note that it can take a few minutes for the UI to reflect the document count and storage size)."
 
 "We have laid the foundation which will allow us to store an **<ins>abitrary</ins>** amount of data in an external Vector Store above and beyond what could fit in memory at the expense of a little more latency."

Thanks!

**Comment by rom212** [comment]
> Hi @rom212, can you please fix the following typos so that the proper checks can run successfully?
> 
> "We can see that our function created an "About Me" index and that our five pieces of **infomation** have been indexed (note that it can take a few minutes for the UI to reflect the document count and storage size)."
> 
> "We have laid the foundation which will allow us to store an **abitrary** amount of data in an external Vector Store above and beyond what could fit in memory at the expense of a little more latency."
> 
> Thanks!

Done in dc8935141b9769b076f792abb8461e3742c3a8ef
Thanks!

**Review by moonbox3 — APPROVED** [review]

**Review by alliscode — APPROVED** [review]

**Pull request merged**